Documentation/Evolution/StringProcessingAlgorithms.md

```diff
@@ -0,0 +1,544 @@
+| `index(sub, start, end)` | `firstIndex(where:)` |  |
+| `join(iterable)` | `reduce(_:_:)` |  |
+| `ljust(width, fillchar)` |  | Alternative: `padding(toLength:)` in Foundation |
+| `lstrip([chars])` |  | |
+| `maketrans(x, y, z)` |  |  |
+| `partition(sep)` |  |  Alternative: `components(separatedBy:)` in Foundation |
+| `removeprefix(prefix)` |  |  |
+| `removesuffix(suffix)` |  |  |
+| `replace(old, new, count)` |  | Alternative: `replacingOccurrences(of:with:)` in Foundation |
+| `rfind(sub, start, end)` | `lastIndex(where:)` | Does not allow specifying a search range |
+| `rindex(sub, start, end)` | `lastIndex(where:)` | Same as above |
+| `rjust(width, fillchar)` |  |  |
+| `rpartition(sep)` |  |  |
+| `rsplit(sep, maxsplit)` |  |  |
+| `rstrip([chars])` |  |  |
+| `split(sep, maxsplit)` | `split(separator:maxSplits:...)` |  |
+| `splitlines(keepends)` | `split(separator:maxSplits:...)` |  |
+| `startswith(prefix, start, end)` | `starts(with:)` |  |  |
+| `strip([chars])` |  | Alternative: `trimmingCharacters(in:)` in Foundation  |
+| `translate(table)` |  |  |
+| `zfill(width)` |  | Alternative: `padding(toLength:)` in Foundation |
+
+
+Note: The comparison table omits functions to query if all characters in the string are of a specified category, such as `isalnum()` and `isalpha()`. These are achievable in Swift by passing in the corresponding character set to `allSatisfy(_:)`, so they're omitted in this table for simplicity. 
+
+### Processing domain-specific strings
+
+Another common task regarding string processing is handling those with domain specific information. 
+
+Consider parsing the date field `"Date: Wed, 16 Feb 2022 23:53:19 GMT"` in an HTTP header as a `Date` type. The naive approach is to search for a substring that looks like a date string (`16 Feb 2022`), and attempt to post-process it as a `Date` with a date parser:
+
+```swift
+let regex = Regex {
+    capture {
+        oneOrMore(.digit)
+        " "
+        oneOrMore(.word)
+        " "
+        oneOrMore(.digit)
+    }
+}
+
+let dateParser = Date.ParseStrategy(format: "\(day: .twoDigits) \(month: .abbreviated) \(year: .padded(4))"
+if let dateMatch = header.firstMatch(of: regex)?.0 {
+    let date = try? Date(dateMatch, strategy: dateParser)
+}
+```
+
+While this approach happens to work for this example, it is fragile when it comes to localized strings. 
+
+Or consider parsing a bank statement to record all the monetary values in the last column:
+
+```swift
+let statement = """
+CREDIT    04/06/2020    Paypal transfer    $4.99
+CREDIT    04/03/2020    Payroll            $69.73
+DEBIT     04/02/2020    ACH transfer       ($38.25)
+DEBIT     03/24/2020    IRX tax payment    ($52,249.98)
+"""
+```
+
+Parsing a currency string such as `$3,020.85` with regex is also tricky, as it can contain localized and currency symbols. This is why Foundation provides industrial-strength parsers for localized strings like these. 
```

Comment:
```suggestion
Parsing a currency string such as `$3,020.85` with regex is also tricky, as it can contain localized and currency symbols in addition to accounting conventions. This is why Foundation provides industrial-strength parsers for localized strings.
```